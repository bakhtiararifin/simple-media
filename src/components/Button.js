import React from 'react'
import { Button } from 'react-native-elements'

import { colors } from '@theme'

const CustomButton = ({ title, ...otherProps }) => {
  return (
    <Button title={title} backgroundColor={colors.primary} color={colors.white} {...otherProps} />
  )
}

export default CustomButton
