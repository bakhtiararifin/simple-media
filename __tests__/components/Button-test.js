import React from 'react'
import Button from '@components/Button'

import renderer from 'react-test-renderer'

test('renders correctly', () => {
  const tree = renderer.create(<Button title="Button" />).toJSON()
  expect(tree).toMatchSnapshot()
})
