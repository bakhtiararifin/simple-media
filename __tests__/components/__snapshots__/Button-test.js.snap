// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 15,
        "marginRight": 15,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <View
      pointerEvents="box-only"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9E9E9E",
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 19,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "backgroundColor": "#1cb5f4",
          },
          undefined,
          Object {
            "padding": 12,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              Object {
                "color": "#fff",
              },
              Object {
                "fontSize": 17.5,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ],
          ]
        }
      >
        Button
      </Text>
    </View>
  </View>
</View>
`;
